plugins/outputs/azure_iothub/iothub/iotdevice/credentials.go

```diff
@@ -0,0 +1,113 @@
+package iotdevice
```

Comment:
In this file the two existing credentials types are implemented. There is another authentication workflow for modules that this code does not implement, but would be a valuable addition. 

Currently the SharedAccessKey credential requires a connection string, or at least a shared access key, in order to authenticate. It does this by creating a token based on the endpoint being accessed (audience), expiry time, and signing it all with the shared access key provided. 

Modules running on the edge can authenticate without the shared access key by creating the token using the same properties but getting it all signed via the workload API /sign endpoint. 

In order to get the token response back from the sign endpoint, the edge runtime is validating the module and generation ID of the running module against what it expects to be running (controlled by the edgeAgent) and so security is provided that way. 

In short, this would allow for telegraf to be deployed without providing a connection string in the configuration. However, some configuration would still be necessary for input/output options.